graphql/schema.graphql

```diff
@@ -1,57 +1,95 @@
-"""User role"""
-enum Role {
-  """Administrator role (department staff etc.)"""
-  ADMIN
-  """User role (professor, student etc.)"""
-  USER
-}
-
 scalar Date
 
 type Query {
-  """Get the current user"""
+  """
+  Get the current user
+  """
   me: User
-  """Find a user by their id"""
+  """
+  Find a user by their id
+  """
   findUserById(id: ID!): User
-  """Get Teaching Preference Survey"""
+  """
+  Get Teaching Preference Survey
+  """
   survey: TeachingPreferenceSurvey!
-  """Get a list of courses for a given term"""
+  """
+  Get a list of courses for a given term
+  """
   courses(term: Term): [CourseSection!]
   """
   Schedule for a given term. If year is given, returns the most recent schedule generated for that year.
   """
   schedule(year: Int): Schedule
+  """
+  Get all courses preferences
+  """
+  coursePreferences: [CoursePreference!]
+  """
+  Get the all users
+  """
+  allUsers: [User!]
 }
 
-"""Generated schedule for a year"""
+"""
+Generated schedule for a year
+"""
 type Schedule {
-  """ID of the schedule"""
+  """
+  ID of the schedule
+  """
   id: ID!
-  """Year for the schedule"""
+  """
+  Year for the schedule
+  """
   year: Int!
-  """When the schedule was generated"""
+  """
+  When the schedule was generated
+  """
   createdAt: Date!
-  """Scheduled courses"""
+  """
+  Scheduled courses
+  """
   courses(term: Term!): [CourseSection!]
 }
 
 type Mutation {
-  """Login into a user account using email and password"""
+  """
+  Login into a user account using email and password
+  """
   login(username: String!, password: String!): AuthPayload!
-  """Logout the currently logged in user"""
+  """
+  Logout the currently logged in user
+  """
```

Comment:
Kinda unrelated, but do we actually need a `logout` mutation? I thought all the frontend would have to do is remove the token from session storage, doesn't need backend logic

Reply:
Since it's cookie-based auth you need to tell the backend to delete the session from the backend session storage.

Reply:
We use it to invalidate the session on backend which seems to be best practice from a security standpoint. It's only 1 line of code.

Reply:
ok. We just won't implement any logic cause it's kinda different on or end

Reply:
I think for the dev, we might need it as we destroy the session in the context or we could just re run the program every time.